softlayer/resource_softlayer_virtual_guest.go

```diff
@@ -417,6 +440,30 @@ func getVirtualGuestTemplateFromResourceData(d *schema.ResourceData, meta interf
 		opts.DedicatedAccountHostOnlyFlag = sl.Bool(dedicatedAcctHostOnly.(bool))
 	}
 
+	if dedicatedHostId, ok := d.GetOk("dedicated_host_id"); ok {
+		opts.DedicatedHost = &datatypes.Virtual_DedicatedHost{
+			Id: sl.Int(dedicatedHostId.(int)),
+		}
+	}
+
+	if dedicatedHostName, ok := d.GetOk("dedicated_host_name"); ok {
+		hostName := dedicatedHostName.(string)
+		service := services.GetAccountService(meta.(ProviderConfig).SoftLayerSession())
+
+		hosts, err := service.
+			Mask("id").
+			Filter(filter.Path("dedicatedHosts.name").Eq(hostName).Build()).
+			GetDedicatedHosts()
+
+		if err != nil {
+			return opts, fmt.Errorf("Error looking up dedicated host '%s': %s", hostName, err)
+		} else if len(hosts) == 0 {
+			return opts, fmt.Errorf("Error looking up dedicated host '%s'", hostName)
+		}
+
+		opts.DedicatedHost = &hosts[0]
+	}
+
```

Comment:
given the restrictions in the schema, shouldn't these 3 if clauses be chained in else-if?

Reply:
Because only one of those 3 options can be set each time we don't actually need chained if-else, but I can do it like that if you think it will make the code more readable.

Reply:
Yes, I think it would make it more explicit as to the mutual exclusivity we are trying to enforce for those options.

Reply:
Fixed.